tests/components/fronius/test_config_flow.py

```diff
@@ -261,3 +267,62 @@ async def test_import(hass, aioclient_mock):
         "host": MOCK_HOST,
         "is_logger": True,
     }
+
+
+async def test_dhcp(hass, aioclient_mock):
+    """Test starting a flow from discovery."""
+    with patch(
+        "homeassistant.components.fronius.config_flow.DHCP_REQUEST_DELAY", 0
+    ), patch(
+        "pyfronius.Fronius.current_logger_info",
+        return_value=LOGGER_INFO_RETURN_VALUE,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": config_entries.SOURCE_DHCP}, data=MOCK_DHCP_DATA
+        )
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["step_id"] == "confirm_discovery"
+
+    result = await hass.config_entries.flow.async_configure(
```

Comment:
Please patch the integration setup function on all create entry results to avoid setting up the whole integration.